lib/services/chain_loader.dart

```diff
@@ -7,15 +7,15 @@ class ChainLoader {
 
   ChainLoader();
 
-  Chain? getChainInfo(int chainId) {
+  Future<Chain> getChainInfo(int? chainId) async {
     if (_chainMap.isEmpty) {
-      _loadChainsAndGenerate();
-      return _chainMap[chainId];
+      await _loadChainsAndGenerate();
+      return Future.value(_chainMap[chainId]);
```

Comment:
This line unconditionally expect chain object and will crash if we don't have particular chain in our database.

Probably we can create some default "placeholder" chain object, which will have real `chainId` and `Unknown` name - in that case, we won't have to handle `null`s.